protocol/rules/byzantine/byzantine.go

```diff
@@ -3,66 +3,47 @@ package byzantine
 
 import (
 	"github.com/relab/hotstuff"
-	"github.com/relab/hotstuff/consensus"
+	"github.com/relab/hotstuff/core"
 	"github.com/relab/hotstuff/modules"
+	"github.com/relab/hotstuff/security/blockchain"
 )
 
-func init() {
-	modules.RegisterModule("silence", func() Byzantine { return &silence{} })
-	modules.RegisterModule("fork", func() Byzantine { return &fork{} })
-}
+const (
+	SilenceModuleName = "silence"
+	ForkModuleName    = "fork"
+)
 
 // Byzantine wraps a consensus rules implementation and alters its behavior.
 type Byzantine interface {
 	// Wrap wraps the rules and returns an altered rules implementation.
-	Wrap(consensus.Rules) consensus.Rules
+	Wrap() modules.ConsensusRules
 }
 
 type silence struct {
-	consensus.Rules
-}
-
-func (s *silence) InitModule(mods *modules.Core) {
-	if mod, ok := s.Rules.(modules.Module); ok {
-		mod.InitModule(mods)
-	}
+	modules.ConsensusRules
 }
 
 func (s *silence) ProposeRule(_ hotstuff.SyncInfo, _ hotstuff.Command) (hotstuff.ProposeMsg, bool) {
 	return hotstuff.ProposeMsg{}, false
 }
 
-func (s *silence) Wrap(rules consensus.Rules) consensus.Rules {
-	s.Rules = rules
+func (s *silence) Wrap() modules.ConsensusRules {
 	return s
 }
 
 // NewSilence returns a byzantine replica that will never propose.
-func NewSilence(c consensus.Rules) consensus.Rules {
-	return &silence{Rules: c}
+func NewSilence(rules modules.ConsensusRules) Byzantine {
```

Comment:
Should return `*silence` since it already embeds the provided `rules`.

Reply:
We should have a test for these Byzantine rules to check that they do what we expect them to do...